app/serializable/serializable_package.rb

```diff
@@ -23,10 +23,10 @@ class SerializablePackage < SerializableResource
     visibility = @object.visibilityData
 
     if visibility['isHidden']
-      { isHidden: true, reason: 'All titles in this package are hidden' }
-    else
-      visibility
+      visibility['reason'] =
+        visibility['reason'] == 'Hidden by EP' ? 'Set by System' : ''
```

Comment:
At package level, if we are modifying the message to "Set by System" or '' , then I do not think we will know at customer resource level, whether all titles in a package are hidden. We need that information to display the message "All titles in this package are hidden" and to make the toggle switch read only in case all titles are hidden at the customer resource level.

Reply:
I believe you can just use the isHidden boolean indicator to know that the titles are all hidden in a package. Logic does not care if it is 'Hidden by EP' or 'Hidden by Customer' in this case.

Reply:
true..shall I just handle this in front-end?

Reply:
hmm - I believe so - but good point since required message is different depending upon whether you are viewing package in context of Package itself or one of it's Package-Titles

Reply:
i agree with carole here - if we aren't already handling this in the UI that's where it should be handled.

Reply:
also, one minor nit here which also applied to @sduvvuri-ebsco 's most recent PR but I didn't catch it then - i _think_ at this point in the execution these are hashes with indifferent access - meaning you may be able to get away with symbol keys (`visibility[:reason]`) instead of string literal keys (`visibility['reason']`).  if that is true, we should change these to use the symbol notation.  it is more [efficient](https://stackoverflow.com/questions/8189416/why-use-symbols-as-hash-keys-in-ruby) and more idiomatic.

Reply:
Thanks for suggestion @sadatay - worked great - did a git commit --amend to make this change. Hopefully ok.  On added message changes - @sduvvuri-ebsco is currently working on changes in this area